src/network/fortinet/fortiauthenticator/restapi/mode/fortitokens.pm

```diff
@@ -0,0 +1,157 @@
+    my $exit = $self->{perfdata}->threshold_check(value => $self->{result_values}->{prct_used}, threshold => [ { label => 'critical-' . $self->{label}, exit_litteral => 'critical' }, { label => 'warning-' . $self->{label}, exit_litteral => 'warning' } ]);
+
+    return $exit;
+}
+
+sub custom_usage_output {
+    my ($self, %options) = @_;
+
+    return sprintf("[tokens ftm] populated:%s - used:%s(%.2f%%) - available:%s(%.2f%%)",
+	    	$self->{result_values}->{populated},
+		$self->{result_values}->{used},
+		$self->{result_values}->{prct_used},
+		$self->{result_values}->{available},
+		$self->{result_values}->{prct_available}
+   );
+}    
+
+
+sub custom_usage_calc {
+    my ($self, %options) = @_;
+
+    $self->{result_values}->{populated} = $options{new_datas}->{$self->{instance} . '_populated'};
+    $self->{result_values}->{available} = $options{new_datas}->{$self->{instance} . '_available'};
+    $self->{result_values}->{used} = $self->{result_values}->{populated} - $self->{result_values}->{available};
+    $self->{result_values}->{prct_used} = $self->{result_values}->{used} * 100 / $self->{result_values}->{populated};
+    $self->{result_values}->{prct_available} = $self->{result_values}->{available} * 100 / $self->{result_values}->{populated};
+
+    return 0;
+}
+
+sub set_counters {
+    my ($self, %options) = @_;
+
+    $self->{maps_counters_type} = [
+        { name => 'fortitokens', type => 0 }
+    ];
+
+    $self->{maps_counters}->{fortitokens} = [
+        { label => 'tokens-usage-prct', set => {
+                key_values => [ { name => 'available' }, { name => 'populated' } ],
+                closure_custom_calc => $self->can('custom_usage_calc'),
+                closure_custom_output => $self->can('custom_usage_output'),
+                closure_custom_perfdata => $self->can('custom_usage_perfdata'),
+                closure_custom_threshold_check => $self->can('custom_usage_threshold'),
+            }
+        },
+    ];
+}
+
+sub new {
+    my ($class, %options) = @_;
+    my $self = $class->SUPER::new(package => __PACKAGE__, %options, force_new_perfdata => 1);
+    bless $self, $class;
+
+    $options{options}->add_options(arguments => {
+    });
+
+    return $self;
+}
+
+sub manage_selection { 
+	my ($self, %options) = @_;
+	my $result = {};
+
+	$self->{fortitokens} = {};
+
+	$result->{populated} = $options{custom}->request_api(
+        endpoint => '/api/v1/fortitokens/?type=ftm&limit=1'
+    );
+    	$result->{available} = $options{custom}->request_api(
+	endpoint => '/api/v1/fortitokens/?type=ftm&limit=1&status=available'
+    );
+
+    $self->{fortitokens} = {
+	available => $result->{available}->{meta}->{total_count},
+	populated => $result->{populated}->{meta}->{total_count}
+    };	   
+
+   foreach my $total_count (values(%{$self->{fortitokens}})) {
+	if (scalar($total_count) <= 0) {
+		$self->{output}->add_option_msg(short_msg => 'API returns empty content.');
+		$self->{output}->option_exit();
+	}
+   }	   
```

Comment:
Counters should be initialized to 0.
Testing emptiness at the end doesn't look good to me.
And ultimately you're deciding that used tokens are those which are not available but from the documentation I see statuses like "new", "pending" or "assigned", so I think we should leverage all those statuses and put them into counters, and maybe create an "available percentage" counter for threshold purpose (opposite of used in a sens so it'll work for your use case).
And what about the FTK tokens ?
Maybe if you could privately provide an API output I could propose a version for you to test.